lib/main.js

```diff
@@ -36,7 +36,7 @@ const DEFAULT_MAX_AGE = 1209600;
  *  },
  * });
  *
- * @returns {Promise<Function>} Resolves with exchange middleware.
```

Comment:
How is this wrong? I think this is the correct way returns {Promise<Type>}

Reply:
The return value is actually an `async` middleware `function` which has not been invoked.  The function will not return a Promise until it is invoked later.  Functions are first class citizens in Javascript which means they can be passed around like any other variable type (string, object, Map, etc).

Reply:
ah okay, thanks for explaining it. @aljones15 sorry, approving it